compiler/rustc_session/src/config.rs

```diff
@@ -1520,7 +1520,7 @@ impl Options {
     }
 
     pub fn get_symbol_mangling_version(&self) -> SymbolManglingVersion {
-        self.cg.symbol_mangling_version.unwrap_or(SymbolManglingVersion::Legacy)
+        self.cg.symbol_mangling_version.unwrap_or(SymbolManglingVersion::V0)
```

Comment:
Nit: Please update `src/doc/rustc/src/symbol-mangling/index.md` as well (says at the bottom that the default is legacy).